v3.0-RC/system_information.json

```diff
@@ -0,0 +1,1227 @@
+          "format": "date"
+        },
+        "privacy_url": {
+          "description":
+          "A fully qualified URL pointing to the privacy policy for the service. An array with one object per supported language with the following keys:",
+          "type": "array",
+          "items": {
+            "type": "object",
+            "properties": {
+              "text": {
+                "description": "The translated text.",
+                "type": "string"
+              },
+              "language": {
+                "description": "IETF BCP 47 language code.",
+                "type": "string",
+                "pattern": "^[a-z]{2,3}(-[A-Z]{2})?$"
+              }
+            },
+            "required": ["text", "language"]
+          }
+        },
+        "privacy_last_updated": {
+          "description":
+          "The date that the privacy policy provided at privacy_url was last updated (added in v2.3-RC).",
+          "type": "string",
+          "format": "date"
+        },
+        "rental_apps": {
+          "description":
+            "Contains rental app information in the android and ios JSON objects (added in v1.1).",
+          "type": "object",
+          "properties": {
+            "android": {
+              "description":
+                "Contains rental app download and app discovery information for the Android platform. (added in v1.1)",
+              "type": "object",
+              "properties": {
+                "store_uri": {
+                  "description":
+                    "URI where the rental Android app can be downloaded from (added in v1.1).",
+                  "type": "string",
+                  "format": "uri"
+                },
+                "discovery_uri": {
+                  "description":
+                    "URI that can be used to discover if the rental Android app is installed on the device (added in v1.1).",
+                  "type": "string",
+                  "format": "uri"
+                }
+              },
+              "required": ["store_uri", "discovery_uri"]
+            },
+            "ios": {
+              "description":
+                "Contains rental information for the iOS platform (added in v1.1).",
+              "type": "object",
+              "properties": {
+                "store_uri": {
+                  "description":
+                    "URI where the rental iOS app can be downloaded from (added in v1.1).",
+                  "type": "string",
+                  "format": "uri"
+                },
+                "discovery_uri": {
+                  "description":
+                    "URI that can be used to discover if the rental iOS app is installed on the device (added in v1.1).",
+                  "type": "string",
+                  "format": "uri"
+                }
+              },
+              "required": ["store_uri", "discovery_uri"]
+            }
+          }
+        }
+      },
+      "anyOf": [
```

Comment:
Here my understanding of the spec is that `license_id` and `license_URL` can both be left blank and that `license_URL` must not be provided if `license_id` is present.

I've tested:
- `license_id` and `license_URL` are present -> invalid ✅
- only`license_id` is present -> valid ✅
- only`license_url` is present -> valid ✅
- `license_id` and `license_URL` are not present- > invalid ❌ (we want this to be valid)

Reply:
I think perhaps `oneOf` might be the proper solution

Reply:
Asking for help from @testower or @AntoineAugusti (or anyone else!), `oneOf` doesn't seem to solve the problem, since it still **requires** one of the options, I can't seem to determine how to represent one of or none. 

Any ideas?

Reply:
From https://stackoverflow.com/a/38799750/2645887

Maybe
```json
"dependencies": {
  "license_url": { "not": { "required": ["license_id"] } },
  "license_id": { "not": { "required": ["license_url"] } }
},
```

Reply:
This seems to work! @isabelle-dr do you want to confirm as well or should I make the change?

Reply:
@josee-sabourin I've just pushed a commit that fixes this problem, please confirm!
It does the following:
- license_id and license_URL are present -> invalid 
- onlylicense_id is present -> valid 
- onlylicense_url is present -> valid 
- license_id and license_URL are not present- > invalid

Reply:
confirmed!